sequence_processing_pipeline/Pipeline.py

```diff
@@ -347,11 +348,41 @@ def get_project_info(self):
 
     @staticmethod
     def is_mapping_file(mapping_file_path):
+        '''
+        Returns true if file is a mapping file.
+        '''
         try:
             Pipeline._validate_mapping_file(mapping_file_path)
             return True
-        except PipelineError:
-            return False
+        except PipelineError as e:
+            # we want to distinguish between a file that is clearly not a
+            # mapping file e.g. sample-file, and a mapping-file that is perhaps
+            # missing a column or has duplicate sample-names.
+            good_messages = ['duplicate sample-names detected:',
+                             'missing columns:']
+            for message in good_messages:
+                if str(e).startswith(message):
+                    return True
+
+        return False
+
+    @staticmethod
+    def is_sample_sheet(sample_sheet_path):
+        '''
+        Returns True if file follows basic sample-sheet format.
+        '''
+
+        # Previously SampleSheet() was used to test if file was a sample-
+        # sheet. SampleSheet() does not return an error on arbitrary-text
+        # files however. Checking to see if the file begins w/[Header]
+        # (ignoring any legacy comments) appears to be the best solution
+        # for now.
+        with open(sample_sheet_path, 'r') as f:
+            lines = f.readlines()
```

Comment:
This is reading all lines but only the first one is being used here, is this necessary?

Reply:
sample-sheets are relatively small and it's a clean way to remove legacy comments at the top of the file but sure I can change it.